File: jja6312/2월 9일/전화번호목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        // 한 번호가 다른 번호의 접두어인가?
        // 만약 indexOf 했을 때 -1이 나온다면? 접두어가 없다. => indexOf가 아닌 startWith이 정답을충족했다.
        // 만약 아니라면 접두어가 있다.
        Arrays.sort(phone_book);
        boolean answer = true;
        for (int i = 0; i < phone_book.length - 1; i++) {
            String thisWord = phone_book[i];
            String nextWord = phone_book[i + 1];
            if (nextWord.startsWith(thisWord)) {
                answer = false;
                break;
            }

        } // for1

        return answer;
    }
}

File: jeong-hyuk/2월9일/전화번호목록.py
def solution(phone_book):
    phone_book.sort()  # 전화번호를 사전순으로 정렬

    for i in range(len(phone_book) - 1):
        # 현재 전화번호가 다음 전화번호의 접두어인지 확인
        if phone_book[i] == phone_book[i + 1][:len(phone_book[i])]:
            return False
    
    return True